velox/functions/prestosql/aggregates/SetAggregates.cpp

```diff
@@ -383,6 +402,61 @@ class SetUnionAggregate : public SetBaseAggregate<T> {
   }
 };
 
+// Returns the number of distinct non-null values in a group. This is an
+// internal function only used for testing.
+template <typename T>
+class CountDistinctAggregate : public SetAggAggregate<T, true> {
+ public:
+  explicit CountDistinctAggregate(
+      const TypePtr& resultType,
+      const TypePtr& inputType)
+      : SetAggAggregate<T, true>(resultType, false), inputType_{inputType} {}
+
+  using Base = SetAggAggregate<T, true>;
+
+  bool supportsToIntermediate() const override {
+    return false;
+  }
+
+  void initializeNewGroups(
+      char** groups,
+      folly::Range<const vector_size_t*> indices) override {
+    exec::Aggregate::setAllNulls(groups, indices);
+    for (auto i : indices) {
+      new (groups[i] + Base::offset_)
+          SetAccumulator<T>(inputType_, Base::allocator_);
+    }
+  }
+
+  void extractAccumulators(char** groups, int32_t numGroups, VectorPtr* result)
```

Comment:
nit: There should be no need to override this since Base::extractAccumulators will also just call extractValues right?

Reply:
Hmm...I've tried but it turns out that we still need to override it here. The reason is that if we don't override, the program will call SetBaseAggregate::extractAccumulator which calls CountDistinctAggregate::extractValues instead of SetBaseAggregate::extractValues.